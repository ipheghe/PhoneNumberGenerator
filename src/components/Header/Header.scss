@import '../../common/functions.scss';

.header__container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: rem(80px);
  box-sizing: border-box;
  width: 100%;
  position: fixed;
  z-index: 10;
  top: 0;
  left: 0;
  right: 0;
  background-color: #57CCCb;
  color: #FFF;
  padding: rem(20px) rem(100px);
  h3 {
    color: #FFF;
    text-transform: uppercase;
  }
  .menu__container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0;
    width: rem(200px);
    max-width: rem(300px);
    .menu__container__items {
      display: flex;
      list-style-type: none;
      text-decoration: none;
      cursor: pointer;
      padding: 0;
      a {
        list-style-type: none;
        text-decoration: none;
        color: #FFF;
        &:hover {
          color: #000;
        }
      }
    }
  }
}

@media screen and (max-width: rem(450px)) {
  .header__container {
    padding: rem(20px) rem(10px);
    h3 {
      font-size: rem(16px);
    }
    .menu__container {
      max-width: rem(100px);
      a {
        font-size: rem(12px);
      }
    }
  }
}
