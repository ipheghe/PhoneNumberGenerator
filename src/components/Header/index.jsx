import React from 'react';
import { Link } from '@reach/router';

import './Header.scss';

/**
 * The Header component primarily used for navigation between pages
 *
 * @returns {JSX} JSX
 */
const Header = () =>  (
  <div className="header__container">
    <h3>Number Generator</h3>
    <ul className="menu__container">
      <li className="menu__container__items">
        <Link to="/">Home</Link>
      </li>
      <li className="menu__container__items">
        <Link to="/operations">Operations</Link>
      </li>
    </ul>
  </div>
);

export default Header;
