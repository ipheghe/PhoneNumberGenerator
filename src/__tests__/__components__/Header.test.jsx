// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import render from 'react-test-renderer';

// components
import Header from '../../components/Header';

describe('Header component', () => {
  it('should match component snapshot', () => {
    const tree = render.create(<Header />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const shallowWrapper = shallow(<Header />);
    expect(shallowWrapper).toBeDefined();
    expect(shallowWrapper.find('.menu__container__items').length).toBe(1);
    expect(shallowWrapper.exists()).toBe(true);
  });
});
