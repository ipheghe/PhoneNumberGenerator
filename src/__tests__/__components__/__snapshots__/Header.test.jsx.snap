// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match component snapshot 1`] = `
<div
  className="header__container"
>
  <h3>
    Number Generator
  </h3>
  <ul
    className="menu__container"
  >
    <li
      className="menu__container__items"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
  </ul>
</div>
`;
